src/nixy.go

```diff
@@ -157,38 +159,96 @@ func setloglevel() {
 	logger.SetLevel(logLevel)
 }
 
-// Eventqueue with buffer of two, because we dont really need more.
-var eventqueue = make(chan bool, 2)
+// set DataManager
+func setupDataManager() {
+	db = *NewDataManager(config.Xproxy, config.LeftDelimiter, config.RightDelimiter, config.MaxFailsUpstream,
+		config.FailTimeoutUpstream, config.SlowStartUpstream)
+	for _, nsConfig := range config.DroveNamespaces {
+		db.CreateNamespace(nsConfig.Name, nsConfig.Drove, nsConfig.User, nsConfig.Pass,
+			nsConfig.AccessToken, nsConfig.Realm, nsConfig.RealmSuffix, nsConfig.RoutingTag, nsConfig.LeaderVHost)
+	}
+}
+
+// Reload signal with buffer of two, because we dont really need more.
+var reloadSignalQueue = make(chan bool, 2)
+var refreshSignalQueue = make(map[string]chan bool)
 
 // Global http transport for connection reuse
 var tr = &http.Transport{MaxIdleConnsPerHost: 10, TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}
 
 func newHealth() Health {
 	var h Health
-	for _, ep := range config.Drove {
-		var s EndpointStatus
-		s.Endpoint = ep
-		s.Healthy = false
-		s.Message = "OK"
-		h.Endpoints = append(h.Endpoints, s)
+	h.NamesapceEndpoints = make(map[string][]EndpointStatus)
+	for _, nsConfig := range config.DroveNamespaces {
+		e := []EndpointStatus{}
+		for _, ep := range nsConfig.Drove {
+			var s EndpointStatus
+			s.Endpoint = ep
+			s.Namespace = nsConfig.Name
+			s.Healthy = false
+			s.Message = "OK"
+			e = append(e, s)
+		}
+		h.NamesapceEndpoints[nsConfig.Name] = e
 	}
 	return h
 }
 
+func setupDefaultConfig() {
+	//default refresh interval
+	if config.EventRefreshIntervalSec <= 0 {
+		config.EventRefreshIntervalSec = 2
+	}
+
+	//default apiTimeout
+	if config.apiTimeout <= 0 {
+		config.apiTimeout = 10
+	}
+}
+
+func setupRefreshChannel() {
+	for _, nsConfig := range config.DroveNamespaces {
+		refreshSignalQueue[nsConfig.Name] = make(chan bool, 2)
+	}
+}
+
+func setupPollEvents() {
+	for _, nsConfig := range config.DroveNamespaces {
+		pollDroveEvents(nsConfig.Name, refreshSignalQueue[nsConfig.Name])
+	}
+}
+
+func setupEndpointHealth() {
+	for _, nsConfig := range config.DroveNamespaces {
+		endpointHealth(nsConfig.Name)
+	}
+}
+
+func forceReload() bool {
+	queued := true
+	for _, nsConfig := range config.DroveNamespaces {
+		select {
+		case refreshSignalQueue[nsConfig.Name] <- true: // Add referesh to our signal channel, unless it is full of course.
+		default:
+			queued = false
+		}
+	}
+	return queued
+}
+
 func nixyReload(w http.ResponseWriter, r *http.Request) {
 	logger.WithFields(logrus.Fields{
 		"client": r.RemoteAddr,
-	}).Info("drove reload triggered")
-	select {
-	case eventqueue <- true: // Add reload to our queue channel, unless it is full of course.
+	}).Info("Reload triggered")
+	queued := forceReload()
```

Comment:
The event queue was present to ensure that reload always happens only from one thread.

Reply:
Done, just renamed